### PR TITLE
Manual merge

### PR DESCRIPTION
**Pull request by phansys**

### CODE REVIEW

File: src/Admin/GalleryAdmin.php [hunk 1]
```diff
@@ -22,7 +22,7 @@
 use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
 
 /**
- * @phpstan-extends AbstractAdmin<\Sonata\MediaBundle\Model\GalleryInterface>
+ * @phpstan-extends AbstractAdmin<\Sonata\MediaBundle\Model\GalleryInterface<\Sonata\MediaBundle\Model\GalleryItemInterface>>
```
Comment (jordisala1991) [inline src/Admin/GalleryAdmin.php:25]: Why on the media admin you added the template but not here? It looks strange

Comment (VincentLanglet) [reply src/Admin/GalleryAdmin.php:25]: And then the MediaAdmin is loosing the template from the BaseMedia so I would say it lost his interest.
Im not sure we should introduce all those generic if noone need them for now.

Comment (phansys) [reply src/Admin/GalleryAdmin.php:25]: I've removed the generic in `MediaAdmin`, as I think we can make that move in a different PR if it is required later.
Thank you for pointing.

**Review by core23 — DISMISSED** [review]

File: src/Model/GalleryInterface.php [hunk 2]
```diff
@@ -15,6 +15,9 @@
 
 use Doctrine\Common\Collections\Collection;
 
+/**
+ * @phpstan-template T of GalleryItemInterface
```
Comment (VincentLanglet) [inline src/Model/GalleryInterface.php:19]: This template is useless if it's not use anywhere else.

Either it should be removed (And if we remove it it's not needed in the BaseGallery, Gallery, ...) or be used in the phpdoc of 
```
public function setGalleryItems(Collection $galleryItems): void;
    public function getGalleryItems(): Collection;
    public function addGalleryItem(GalleryItemInterface $galleryItem): void;
    public function removeGalleryItem(GalleryItemInterface $galleryItem): void;
```

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**